src/fu-engine-requirements.c

```diff
@@ -11,37 +11,36 @@
 #include "fu-engine-requirements.h"
 
 static gboolean
-fu_engine_requirements_require_vercmp(XbNode *req,
-				      const gchar *version,
-				      FwupdVersionFormat fmt,
-				      GError **error)
+fu_engine_requirements_require_vercmp_part(const gchar *compare,
+					   const gchar *version_req,
+					   const gchar *version,
+					   FwupdVersionFormat fmt,
+					   GError **error)
 {
 	gboolean ret = FALSE;
 	gint rc = 0;
-	const gchar *tmp = xb_node_get_attr(req, "compare");
-	const gchar *version_req = xb_node_get_attr(req, "version");
```

Comment:
I wonder if it's worth introducing a new attribute `compare_glob` to make this less painful to backport and let earlier versions not care if they're missing it?